### ISSUE TITLE
AQL grammar online

### ISSUE THREAD

**Issue by mingodad** [arangodb/arangodb#19774]
I've just added this project `AQL` grammar on https://mingodad.github.io/parsertl-playground/playground/ that is an editor/tester for yacc/lex grammars, select `ArangoDB AQL parser` from `Examples` then click `Parse` to see the parser tree of the `aql` in `Input source`.

I hope it'll help document/debug/develop/test the grammar .

Any feedback is welcome !

**Comment by dothebart** [comment]
hi, 
is this related to https://github.com/arangodb/arangodb/issues/1755 ?

**Comment by mingodad** [comment]
No, it's more grmmar yacc/lex related than highlighting support for editors, here a list of grammars so far with varying degree of functionality correctness:
- Ada parser
- Akwa parser (not working)
- Aliceml parser (partially working)
- Ansi C11 parser (partially working)
- Ansi C18 parser (partially working)
- Ansi C parser
- Antlr4.5 parser
- ArangoDB AQL parser
- Austral parser (partially working)
- Bison parser (partially working)
- BNFGen parser
- Bolt parser
- Blawn parser
- Braille parser (not working)
- C3c  parser
- Calculator parser
- Carbon parser (need review of '*')
- Chapel parser
- CocoR parser (partially working)
- CoqPP parser
- Cpp5-v2 parser (not working)
- Cql parser
- C parser frama-c (partially working)
- Cxx parser (not working)
- D parser (partially working)
- Doxygen code scanner torture
- Ecere parser (not working)
- Expr-lang parser
- GramGrep parser
- HTML parser
- idl2cpp parser
- Ispc parser
- Java11 parser
- JavascriptCore parser
- Jq parser (partially working)
- Json5 parser
- Json lexer
- Json parser
- LALR parser
- Lark parser
- Lezer parser (partially working)
- LFortran parser (partially working)
- Linden Script parser
- Lox parser
- LPegrex parser (partially working)
- LPython parser (not working)
- Lua2ljs parser
- Lua-5.3 parser
- Lua parser
- LuaPP parser (partially working)
- Menhir stage1 parser
- Menhir fancy parser
- Minic parser
- Minizinc parser (not working)
- MSTA parser
- Mulang parser (not working)
- Mysql parser (be patient)(partially working)
- OcamlLex parser (partially working)
- Ocaml parser from rescript (partially working)
- Openscad parser (partially working)
- Peg parser (partially working)
- PHP-8.2 parser (partially working)
- Pikchr parser
- Playground3 parser
- Playground parser
- PnetC parser
- PnetCSHarp parser
- PnetDPas parser
- PnetJava parser
- PnetVBasic parser
- Postgresql parser (be patient)
- Preprocessor parser (not working)
- Protobuf3 parser
- R parser (partially working)
- ReasonML parser (partially working)
- Rust parser
- SC-im spreadsheet parser
- Scheme parser
- Souffle parser
- SQLite3 parser (partially working)
- Tameparse parser (not working)
- TDengine SQL parser
- Textdiagram parser
- Textmapper parser
- Webassembly interpreter parser
- XML parser
- Z80 assembler parser

**Comment by dothebart** [comment]
so it should visualize the parsing process to to the user?

**Comment by mingodad** [comment]
The parser tree specifically, but it's more for grammar writers than non programmer users.

**Comment by jsteemann** [comment]
@mingodad : thanks for the addition. I noticed that when copy & pasting the generated ENBF from ParserTL playground by clicking into the window part with the generated ENBF, pressing CTRL + A (to select all) and then CTRL + C (to copy everything), the clipboard also contained slightly more stuff (see the orange-highlighted parts in the attached screenshot) that I had to manually remove when pasting the result into some other program.
![image](https://github.com/arangodb/arangodb/assets/1423118/f4744244-7489-4a29-a3d2-a3b302a8bb5b)
I think this could be improved by making the generated output go into a proper (readonly) textarea or so, so that when selecting & copying its contents, you only mark & copy the actually relevant stuff.
Thanks!

**Comment by mingodad** [comment]
Thank you for your feedback !
There are several things that can be made better, right now I'm using the same a `div` to output the `EBNF/YACC` for simplicity because it's the same place that shows info about the grammar and errors are links to make the editor jump to the error line.

I did a small change to make a bit easier to use the `EBNF/YACC` generated code, now I try to copy it directly to the clipboard through `Javascript`.

Try again and you'll see that the `Generate` select box now shows `EBNF grammar and copy to clipboard`.

Cheers !
